False positive on POSIX sh with export -p

#### For bugs
- Rule Id: SC2039
- Shellcheck version 0.6.0 and online. 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2039)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
export -p
```

#### Here's what shellcheck currently says:
```
Line 2:
export -p
       ^-- SC2039: In POSIX sh, export -p is undefined.
```

#### Here's what I wanted or expected to see:
Nothing, `-p` is the only valid flag for export in POSIX sh ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#export)). This false positive was introduced when #1130 was fixed.

--- a/src/ShellCheck/Checks/ShellSupport.hs
+++ b/src/ShellCheck/Checks/ShellSupport.hs
@@ -141,6 +141,8 @@ prop_checkBashisms58= verify checkBashisms "#!/bin/sh\nulimit -c 0"
 prop_checkBashisms59 = verify checkBashisms "#!/bin/sh\njobs -s"
 prop_checkBashisms60 = verifyNot checkBashisms "#!/bin/sh\njobs -p"
 prop_checkBashisms61 = verifyNot checkBashisms "#!/bin/sh\njobs -lp"
+prop_checkBashisms62 = verify checkBashisms "#!/bin/sh\nexport -f foo"
+prop_checkBashisms63 = verifyNot checkBashisms "#!/bin/sh\nexport -p"
 
 checkBashisms = ForShell [Sh, Dash] $ \t -> do
     params <- ask
@@ -286,7 +288,7 @@ checkBashisms = ForShell [Sh, Dash] $ \t -> do
             ] ++ if not isDash then ["local"] else []
         allowedFlags = Map.fromList [
             ("exec", []),
-            ("export", ["-p"]),
+            ("export", ["p"]),
             ("jobs", ["l", "p"]),
             ("printf", []),
             ("read", if isDash then ["r", "p"] else ["r"]),

